Where2Play/Pages/CitySearch.cshtml.cs

```
@@ -46,8 +48,15 @@ public async Task OnPostAsync()
 
             try
             {
+                var apiKey = _configuration["ApiKeys:SetlistFm"];
+                if (string.IsNullOrEmpty(apiKey))
+                {
+                    Console.WriteLine("Setlist.fm API key not configured.");
+                    return;
+                }
+
```

Retrieved API Key from Configuration.
This removes the hardcoded API key and reads it from configuration instead. It also adds validation to ensure the key is configured before attempting API calls, providing fail-fast behavior.

Best Practice & Source:
•  OWASP - Cryptographic Storage: Never hardcode secrets in source code as they can be exposed through version control
◦  Source: https://cheatsheetseries.owasp.org/cheatsheets/Cryptographic_Storage_Cheat_Sheet.html
•  Defensive Programming: Validate inputs and configuration early to fail fast with clear error messages
◦  Source: McConnell, Steve. "Code Complete 2" (2004)
•  Microsoft Documentation: "Don't hardcode passwords, connection strings, or other sensitive data into your application"
◦  Source: https://learn.microsoft.com/en-us/dotnet/standard/security/secure-coding-guidelines

---

Very good.